usr/src/lib/libdladm/common/libdladm.c

```diff
@@ -142,6 +145,12 @@ dladm_dld_fd(dladm_handle_t handle)
 	return (handle->dld_fd);
 }
 
+kstat_ctl_t*
+dladm_dld_kctl(dladm_handle_t handle)
```

Comment:
You've defined this function, I think you should use it wherever you currently have `handle->kcp` in the other code. This is how the existing `dladm_dld_fd()` is used.